src/System.Diagnostics.TraceSource/tests/CorrelationManagerTests.cs

```diff
@@ -0,0 +1,136 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+using System.Collections;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Diagnostics.TraceSourceTests
+{
+    public class CorrelationManagerTests
+    {
+        [Fact]
+        public void CorrelationManager_CheckStack()
+        {
+            Trace.CorrelationManager.StartLogicalOperation("one");
+            Trace.CorrelationManager.StartLogicalOperation("two");
+
+            // 2 operations in progress
+            ValidateStack(Trace.CorrelationManager.LogicalOperationStack, "two", "one");
+
+            Trace.CorrelationManager.StopLogicalOperation();
+            // 1 operation in progress
+            ValidateStack(Trace.CorrelationManager.LogicalOperationStack, "one");
+
+            Trace.CorrelationManager.StopLogicalOperation();
+            // 0 operation in progress
+            ValidateStack(Trace.CorrelationManager.LogicalOperationStack, Array.Empty<object>());
+        }
+
+        [Fact]
+        public void CorrelationManager_NullOperationId()
+        {
+            Assert.Throws<ArgumentNullException>("operationId", () => Trace.CorrelationManager.StartLogicalOperation(null));
+        }
+
+        [Fact]
+        public void CorrelationManager_EmptyStack()
+        {
+            Assert.Throws<InvalidOperationException>(() => Trace.CorrelationManager.StopLogicalOperation());
+        }
+
+        [Fact]
+        public void CorrelationManager_ActivityId()
+        {
+            Guid id = Guid.NewGuid();
+            Trace.CorrelationManager.ActivityId = id;
+
+            Assert.Equal(id, Trace.CorrelationManager.ActivityId);
+        }
+
+        [Fact]
+        public void CorrelationManager_MutateStackPush()
+        {
+            Guid id = Guid.NewGuid();
+            Trace.CorrelationManager.ActivityId = id;
+
+            Trace.CorrelationManager.StartLogicalOperation(1);
+
+            Task.Yield().GetAwaiter().GetResult();
```

Comment:
This is a nop.  In this and other cases like it, you should make the test itself an "async Task" instead of "void" method and then "await Task.Yield()".